src/rgw/rgw_sal_motr.cc

```diff
@@ -2504,6 +2595,63 @@ int MotrAtomicWriter::process(bufferlist&& data, uint64_t offset)
   return this->write();
 }
 
+void MotrObject::update_null_version_index(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
+{
+  int rc = 0;
+  string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
+  string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
+  std::string null_version_key = this->get_name() + "/null";
+  bufferlist::const_iterator iter;
+  bufferlist bl, bl_null_idx_val;
+  rgw_bucket_dir_entry current_null_key;
+  current_null_key.key.name = ent.key.name;
+  current_null_key.key.instance = ent.key.instance;
+  current_null_key.encode(bl_null_idx_val);
+
+  // TODO : cache
+  // GET an extra entry
+  rc = this->store->do_idx_op_by_name(bucket_index_iname,
+                              M0_IC_GET, null_version_key, bl);
+  ldpp_dout(dpp, 20) <<__func__<< "GET null rc : " << rc << dendl;
+
+  if(rc == 0)
+  {
+    rgw_bucket_dir_entry null_key;
+    bufferlist& blr = bl;
+    iter = blr.cbegin();
+    null_key.decode(iter);
+    std::string prev_null_key = null_key.key.name + '[' + null_key.key.instance + ']';
+    // delete prev null object entry
+    // clear the bufferlist -> obj1[null] = {}
+    bl.clear();
+    rc = this->store->do_idx_op_by_name(bucket_index_iname,
+                M0_IC_DEL, prev_null_key, bl);
+    ldpp_dout(dpp, 20) <<__func__<< "DELETE null rc : " << rc << dendl;
+    bl.clear();
+    rc = this->store->do_idx_op_by_name(bucket_index_iname,
+                M0_IC_DEL, null_version_key, bl);
+    ldpp_dout(dpp, 20) <<__func__<< "DELETE null rc : " << rc << dendl;
+
+    // obj1[null] = {current}
+    rc = this->store->do_idx_op_by_name(bucket_index_iname,
+                              M0_IC_PUT, null_version_key, bl_null_idx_val);
+    ldpp_dout(dpp, 20) <<__func__<< "PUT after DELETE null rc : " << rc << dendl;         
+  }
+
+  // if null entry not present in the motr (rc<0), 
+  // then add the new null index entry with latest/current
+  // version-id reference.
+  if(rc < 0)
```

Comment:
this should 'else' case for 'if' of line 2617, else it would execute based on rc value if PUT on line 2636 failed too. or is that the intention, and comments above are misleading ?

Reply:
This extra if statement is required and added intentionally to create a new null entry for the very first time when it doesn't exists. 
Updated the comment.